doc/show.in.man
.\" Originally generated by cmd.
.TH SHOW "1" "Monday July 2014" "show 2014.06.07" "User Commands"
.SH NAME
show \- Show Fun
.SH SYNOPSIS
.B show [OPTIONS] [SHOW] [-- SHOWOPTIONS]
.SH EXAMPLE
.B show cookie
.SH DESCRIPTION
shell show is a command that displays nice graphical shell art. shell show has various effects combined.
.PP
Mandatory arguments for long options are mandatory for short options too.
.TP
\fB\-l\fR, \fB\-\-list\fR
list shows
.TP
\fB\-s\fR, \fB\-\-sound\fR
play a sound
.TP
\fB\-f\fR, \fB\-\-foreground\fR
work in foreground
.TP
\fB\-g\fR, \fB\-\-gingerup\fR
ginger up for data
.TP
\fB\-v\fR, \fB\-\-version\fR
show the program version
.TP
\fB\-v\fR, \fB\-\-variable\fR=\fIVARIABLE\fR
show a program variable
.TP
\fB\-h\fR, \fB\-\-help\fR
show the program help
.TP
\fB\-m\fR, \fB\-\-man\fR
show the program manual
.TP
By default shows a cookie.
.TP
The available --variable VARIABLES are: package, name, version, description, explanation, license, homepage, author, email, social, blog, usage, options, examples, extrahelp, extranotes, month, year, day and timestamp.
.TP
Execute 'bash -c "man <(show --man)"' to see the runtime manpage.
.SH NOTES
For more information, check documentation.
.SH HOMEPAGE
Main project page at \fBhttp://code.google.com/p/codemiscs\fR.
.SH AUTHOR
Programmed by \fBJuan Manuel Borges Caño\fR.
.SH EMAIL
Thank, Contact, Follow through \fBjuanmabcmail@gmail.com\fR.
.SH SOCIAL
Interact with \fBhttp://plus.google.com/+JuanManuelBorgesCaño\fR
.SH SUPPORT
Support through Paypal, Patreon, Flattr, Western Union, BitCoin, ...
.SH COPYRIGHT
Copyright \(co \fB2014 Juan Manuel Borges Caño\fR.
.PP
This software is licensed and under the terms of the GPLv3+ License.
There is NO WARRANTY, to the extent permitted by law.


NEWS
2014-07-03 Juan Manuel Borges Caño <juanmabcblog.blogspot.com, juanmabcmail@gmail.com> 
	* New Commands: tonize, status.
	* Improved Commands: colorize, tw.
	* Improved Website: better looking design with more easy access to stuff that matters, and more content like NEWS & ChangeLog.
	* Better Design: colorize understands pattern and line matches, removes reverse duplication, tw adapts way faster to online engines changes. Website makes core usage of templates & is better metatagged.
	* Better Documentation: continues improving effortlessly through documentation automatic syndication, automatic tonize & status man pages.
	* Better Distribution: more automated generation of documentation and wiki.
	* Better Motivation: proved benefits sweetens development, nicer web with a lot of content exploit.

2014-03-16 Juan Manuel Borges Caño <juanmabcblog.blogspot.com, juanmabcmail@gmail.com> 
	* Proved Benefits: previous features consolidate actual ones.
	* Better Motivation: proved benefits sweetens development.
	* Improved Commands: cmd (core), typewrite, colorize.
	* Improved Documentation: more rich content with media.
	* Improved Distribution: more automated generation of binary packages.
	* Better Design: typewrite can scroll, and pass through non printable characters, colorize understands more keywords.
	* Better Website: thanks to better documentation and better distribution.

2014-02-10 Juan Manuel Borges Caño <juanmabcblog.blogspot.com, juanmabcmail@gmail.com> 
	* Improved Commands: cmd (core), tw, show.
	* Better Website: Continues improving effortlessly through documentation automatic syndication and smart meta distribution. Demonstrations of every command.
	* Better Design: The cmd improvements benefit instantly all commands coding, look and usage experiences. Show plugin design is more modulable and easily expandable.
	* Better Documentation: Continues improving effortlessly through documentation automatic syndication. Code parchments got everywhere where matters.
	* Better Distribution: Introducing smart meta distribution technology (currently targeted to open build service, build.opensuse.org instance, supporting fedora, ubuntu, suse, debian, centos, arch and autotools).
	* Proved Benefits: previous releases features consolidate into real value.
	* Better Motivation: copyrighted for 2014.

2013-12-30 Juan Manuel Borges Caño <juanmabcblog.blogspot.com, juanmabcmail@gmail.com> 
	* New Commands, Improved Commands, Better Design, Better Documentation, Better Motivation.

2013-12-19 codemiscs <code.google.com/p/codemiscs>
	* New Commands, Better Documentation, Better Website, Better Distribution.

2013-11-28 codemiscs <code.google.com/p/codemiscs>
	* Independent Projects Merger.


cmd/show/show.in.sh
#! /usr/bin/env bash

#  _________________________________________________________________________
# /\                                                                        \
# \_|       ___        __                            _   _                  |
#   |      |_ _|_ __  / _| ___  _ __ _ __ ___   __ _| |_(_) ___  _ __       |
#   |       | || '_ \| |_ / _ \| '__| '_ ` _ \ / _` | __| |/ _ \| '_ \      |
#   |       | || | | |  _| (_) | |  | | | | | | (_| | |_| | (_) | | | |     |
#   |      |___|_| |_|_|  \___/|_|  |_| |_| |_|\__,_|\__|_|\___/|_| |_|     |
#   |                                                                       |
#   |               show (Shell Show): Show Fun (Early stages)              |
#   |            Copyright (C) 2013 - 2014 Juan Manuel Borges Caño          |
#   |                      Terminal graphics amaze me.                      |
#   |      Some shows are based on the gathering of http://mewbies.com/     |
#   |                    _     _                                            |
#   |                   | |   (_) ___ ___ _ __  ___  ___                    |
#   |                   | |   | |/ __/ _ \ '_ \/ __|/ _ \                   |
#   |                   | |___| | (_|  __/ | | \__ \  __/                   |
#   |                   |_____|_|\___\___|_| |_|___/\___|                   |
#   |                                                                       |
#   |  This program is free software: you can redistribute it and/or modify |
#   |  it under the terms of the GNU General Public License as published by |
#   |   the Free Software Foundation, either version 3 of the License, or   |
#   |                  (at your option) any later version.                  |
#   |                                                                       |
#   |    This program is distributed in the hope that it will be useful,    |
#   |     but WITHOUT ANY WARRANTY; without even the implied warranty of    |
#   |     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the     |
#   |              GNU General Public License for more details.             |
#   |                                                                       |
#   |   You should have received a copy of the GNU General Public License   |
#   | along with this program.  If not, see <http://www.gnu.org/licenses/>. |
#   |   ____________________________________________________________________|_
#    \_/______________________________________________________________________/


shopt -s extglob

# The --list option
function sh_option_list
{
	for sh_show in "$cmd_datadir/shows"/*.sh
	do
		sh_show="$sh_show"
		sh_show="${sh_show##*/}"
		sh_show="${sh_show##*_}"
		sh_show="${sh_show%.*}"
		printf "%s\n" "${sh_show}"
	done
	exit 0
}

# The --sound option
function sh_option_sound
{
	sh_sound="on"
}

# The --foreground option
function sh_option_foreground
{
	sh_foreground="on"
}

# ... and show, the program itself.

# The cmd init function
function sh_init
{
	shopt -s extglob
	shopt -s nullglob

	sh_sound="off"
	sh_foreground="off"
	#sh_show="cookie"
}

# The cmd main function
function sh_main
{
	# Select user show or random
	if [[ -n "$1" ]]
	then
		sh_show="$1"
		shift
	else
		sh_shows=("$cmd_datadir/shows"/*.sh)
		sh_show="${sh_shows[$((${RANDOM}%${#sh_shows[@]}))]}"
		sh_show="${sh_show##*/}"
		sh_show="${sh_show##*_}"
		sh_show="${sh_show%.*}"
	fi

	# Show Time!
	if [[ -e "${cmd_datadir}/shows/${sh_show}.sh" ]]
	then
		source "${cmd_datadir}/shows/${sh_show}.sh" "$@"
	else
		cmd_error "unknown show"
	fi
}

# The cmd fields
cmd_package="[@]pkg[@]"
cmd="show"
cmd_name="shell show"
cmd_version="[@]pkgversion[@]"
cmd_description="Show Fun"
cmd_explanation="shell show is a command that displays nice graphical shell art. shell show has various effects combined."
cmd_license="[@]pkglicense[@]"
cmd_homepage="[@]pkghomepage[@]"
cmd_author="[@]pkgauthor[@]"
cmd_blog="[@]pkgblog[@]"
cmd_email="[@]pkgemail[@]"
cmd_usage="$cmd [OPTIONS] [SHOW] [-- SHOWOPTIONS]"
cmd_options=("/l/list/list shows/sh_option_list/" "/s/sound/play a sound/sh_option_sound/" "/f/foreground/work in foreground/sh_option_foreground/")
cmd_examples=("$cmd cookie")
cmd_extrahelp="By default shows a cookie."
cmd_extranotes="For more information, check documentation."
cmd_init="sh_init"
cmd_main="sh_main"

cmd_datadir="[@]pkgdatadir[@]/$cmd"

# The cmd environment
source "[@]pkgdatadir[@]/cmd.sh"
